src/main/java/org/openapitools/api/impl/SessionsApiServiceImpl.java

```diff
@@ -23,6 +25,30 @@ public SessionsApiServiceImpl(@Nonnull Identity identity) {
 
     @Override
     public Response sessionsPost(Session session, SecurityContext securityContext) throws NotFoundException {
-        return Response.ok().entity(new ApiResponseMessage(ApiResponseMessage.OK, "magic!")).build();
+        try {
+            SessionCreate sessionCreate = identity.getSessionService().newSessionCreate();
+            sessionCreate.setUsername(session.getUsername());
+            sessionCreate.setPassword(session.getPassword());
+            sessionCreate.initCredential();
+
+            AbuseDetectionImpl abuseDetectionImpl = new AbuseDetectionImpl(identity);
```

Comment:
Abuse detection should be abstracted away in the core logic.
We shouldn't need to explicitly call `abuseDetection` here.